config/config.go

```diff
@@ -1018,6 +1056,27 @@ func migrateConfigPurposeOneTreatment(v *viper.Viper) {
 	}
 }
 
+func migrateConfigTCF2PurposeEnabledFlags(v *viper.Viper) {
+	for i := 1; i <= 10; i++ {
+		oldField := fmt.Sprintf("gdpr.tcf2.purpose%d.enabled", i)
+		newField := fmt.Sprintf("gdpr.tcf2.purpose%d.enforce_purpose", i)
+
+		if v.IsSet(oldField) {
+			oldConfig := v.GetBool(oldField)
+			if v.IsSet(newField) {
+				glog.Warningf("using %s and ignoring deprecated %s", newField, oldField)
+			} else {
+				glog.Warningf("%s is deprecated and should be changed to %s", oldField, newField)
+				if oldConfig == true {
```

Comment:
Nitpick: There is no need to add `== true` to a conditional. This will generate a linter warning.